### ISSUE TITLE
Enhance tag search to use fulltext search

### ISSUE THREAD

**Issue by MrGeneration** [zammad/zammad#2569]
### Infos:

* Used Zammad version: 2.9
* Installation method (source, package, ..): any
* Operating system: any
* Database + version: any
* Elasticsearch version: any
* Browser + version: any
* Ticket-ID: #1042889

### Expected behavior:

Searching for a substring of a tag will work.
Example, if your tag is `feedback`, you can also search for `back` and the search result will be `feedback`.

### Actual behavior:

Zammad currently only searches `LIKE {search-string}*` which makes it impossible to find substrings or parts of tags. This especially kicks in, if you only remember or know a part of the tag name.

### Steps to reproduce the behavior:

* create a tag e.g. `feedback`
* open any ticket and click on "Add Tag"
* now search for a sub string of the tag, e.g. `back` - you will receive no results

### Note to developers:

This is rather a small change (https://github.com/zammad/zammad/blob/d4b0d953d00faa515fb5cf6cdb159a696715c9f0/app/controllers/tags_controller.rb#L8), but will require tests to ensure the bevahiour keeps as expected (and stops us from patching it out by accident :-P )

Basically from
`list = Tag::Item.where('name_downcase LIKE ?', "#{params[:term].strip.downcase}%").order(name: :asc).limit(params[:limit] || 10)` 
to
`list = Tag::Item.where('name_downcase LIKE ?', "%#{params[:term].strip.downcase}%").order(name: :asc).limit(params[:limit] || 10)`

**Comment by dennykorsukewitz** [comment]
hi @martini is that already solved?

**Comment by martini** [comment]
Is already fixed in 3.1 tree.
